pkg/apis/operator/v1alpha1/tektonresult_types.go

```diff
@@ -83,6 +83,9 @@ type ResultsAPIProperties struct {
 	S3SecretAccessKey     string `json:"s3_secret_access_key,omitempty"`
 	S3MultiPartSize       int64  `json:"s3_multi_part_size,omitempty"`
 	LoggingPVCName        string `json:"logging_pvc_name"`
+	// name of the secret used to get S3 credentials and
+	// pass it as environment variables to the "tekton-results-api" deployment under "api" container
+	SecretName string `json:"secret_name"`
```

Comment:
I think we should start using a string pointer for a new field, because in case in future if we don't need that field, we can set the field directly to nil in defaulting and will also help in upgrade

Reply:
Also I think we should add omitempty in the json tag

Reply:
added `omitempty`